crates/pathfinder/src/bin/pathfinder/config.rs

```diff
@@ -205,6 +205,34 @@ This should only be enabled for debugging purposes as it adds substantial proces
         env = "PATHFINDER_GATEWAY_API_KEY"
     )]
     gateway_api_key: Option<String>,
+
+    #[arg(
+        long = "storage.event-bloom-filter-cache-size",
+        long_help = "The number of blocks whose event bloom filters are cached in memory. \
+            This cache speeds up event related RPC queries at the cost of using extra memory. \
+            Each cached filter takes 2 KiB of memory.",
+        env = "PATHFINDER_STORAGE_BLOOM_FILTER_CACHE_SIZE",
+        default_value = "524288"
+    )]
+    event_bloom_filter_cache_size: std::num::NonZeroUsize,
+
+    #[arg(
+        long = "rpc.get-events-max-blocks-to-scan",
+        long_help = "The number of blocks to scan for events when querying for events. \
+            This limit is used to prevent queries from taking too long.",
+        env = "PATHFINDER_RPC_GET_EVENTS_MAX_BLOCKS_TO_SCAN",
+        default_value = "500"
+    )]
+    get_events_max_blocks_to_scan: std::num::NonZeroUsize,
+
+    #[arg(
+        long = "rpc.get-events-max-bloom-filters-to-load",
+        long_help = "The number of Bloom filters to load for events when querying for events. \
+            This limit is used to prevent queries from taking too long.",
+        env = "PATHFINDER_RPC_GET_EVENTS_MAX_BLOOM_FILTERS_TO_LOAD",
+        default_value = "100000"
+    )]
+    get_events_max_bloom_filters_to_load: std::num::NonZeroUsize,
```

Comment:
Okay having read the source code I now understand. The scanning (former) refers to actually opening a block and looking inside, the bloom filter loading (latter) refers to the number of bloom filters to load and check.

Since we expect each block will have a bloom filter, that effectively means that the latter is equivalent to the maximum block range per result?

I'm hoping we can come up with something that is understandable to a lay-person that knows a little about bloom filters.

Maybe the former can be something like max-bloom-filter-matches?

Reply:
> Since we expect each block will have a bloom filter, that effectively means that the latter is equivalent to the maximum block range per result?

Not exactly: cached Bloom filters are not counted here. (The idea is that the check takes significantly less time than loading the filter.)

I can understand this all is a bit complicated since understanding what these mean actually require understanding what's going on during evaluation...

Reply:
Ah I see. Hmm. max-uncached-bloom-filters?

Reply:
That seems like a correct (if long) description of it.

Reply:
That sounds like a better name to me.